src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java

```diff
@@ -51,6 +53,18 @@ public int getConvictionCount() {
             return convictionCount;
         }
     }
+    public static class DuplicateConvictionsForSentenceDateException extends Exception {
```

Comment:
This is for my benefit really.

Lots of checked exceptions for controlling flow?  Just seems a bit old school.  Is it historic?  Sorry don't know what is new versus existing...

Reply:
Good point; we have 3 states; 

1. I have found the thing
2. I have not found the thing
3. I have found loads of things when there should only be 1.

Not sure the best way to handle that other than am Optional and exception.

I suppose I could have a `Result` type class instead. Any better ideas @matt-crowson-moj

Reply:
If only java had such a class (Kotlin caters this one).  Of course there are 3rd party libs that support this sort of thing providing Try and Result objects...